admin-ui/plugins/auth-server/redux/api/ScriptApi.js

```diff
@@ -0,0 +1,66 @@
+import { handleResponse } from 'Utils/ApiUtils'
+
+export default class ScriptApi {
+  constructor(api) {
+    this.api = api
+  }
+
+  // Get all scripts
+  getAllScripts = (options = {}) => {
+    return new Promise((resolve, reject) => {
+      this.api.getConfigScripts(options, (error, data) => {
+        handleResponse(error, reject, resolve, data)
+      })
+    })
+  }
+
+  // Get scripts by type
+  getScriptsByType = (scriptType, options = {}) => {
+    return new Promise((resolve, reject) => {
+      // Use fieldValuePair to filter by scriptType
+      const searchOptions = {
+        ...options,
+        fieldValuePair: `scriptType:${scriptType}`,
+      }
+      this.api.getConfigScripts(searchOptions, (error, data) => {
```

Comment:
Unconditionally overwriting the `fieldValuePair` property and mutating the `options` object in-place is bad because it violates the principle of immutability by directly modifying the `options` argument passed into the function instead of creating a new, separate options object. This creates a side effect that can lead to unpredictable behavior in the calling code, which does not expect its data to be altered.